src/utils/ingest-client/ingest-file-validator.ts

```diff
@@ -30,22 +31,39 @@ class IngestFileValidator {
      * @param fileName
      */
     validateFile(fileResource: any, fileFormat: string, fileName: string) : Promise<ValidationJob> {
-        return this.assertNotAlreadyValidated(fileResource).then(fileChecksums => {
-            return this.uploadAreaForFile(fileResource).then(uploadAreaUuid => {
-                const validationImage = this.imageFor(fileFormat);
-                if(! validationImage) {
-                    return Promise.reject(new NoFileValidationImage());
-                } else {
-                    const imageUrl = validationImage.imageUrl;
-                    return IngestFileValidator._validateFile(fileName, uploadAreaUuid, imageUrl, this.uploadClient).then(validationJobId => {
-                        return Promise.resolve({
+
+        const validationJob = fileResource["validationJob"];
+        const fileChecksums = fileResource["checksums"];
+
+        if (validationJob) {
+            const completed = validationJob.jobCompleted;
+            if (!completed) {
+                return Promise.reject(new FileCurrentlyValidatingError());
+            }
+
+            if (validationJob.validationReport) {
+                const fileSha1 = fileChecksums.sha1;
+                const validatedSha1 = validationJob.checksums.sha1;
+
+                if (validationJob.validationReport.validationState == 'VALID' && fileSha1 == validatedSha1)
+                    return  Promise.resolve(validationJob);
+            }
+        }
+        return this.uploadAreaForFile(fileResource).then(uploadAreaUuid => {
+            const validationImage = this.imageFor(fileFormat);
+            if(! validationImage) {
+                return Promise.reject(new NoFileValidationImage());
+            } else {
+                const imageUrl = validationImage.imageUrl;
+                return IngestFileValidator._validateFile(fileName, uploadAreaUuid, imageUrl, this.uploadClient).then(validationJobId => {
+                    return Promise.resolve({
                             validationId: validationJobId,
                             checksums: fileChecksums,
-                            jobCompleted: false
-                        });
+                            jobCompleted: false,
+                            validationJob: null // reset validationJob
```

Comment:
Should this be `validationReport: null`?

Reply:
You're right. I don't think it's actually needed to be set anyway.

Reply:
OK, it's actually needed to reset any previous job report. Updated now.